src/main.cpp

```diff
@@ -33,1689 +26,29 @@
-          }
-          ret = true;
-        }
-      }
-      else
-        streamReader->Reset(true);
-    }
-  }
-
-  return ret;
-}
-
-void Session::OnSegmentChanged(adaptive::AdaptiveStream* adStream)
-{
-  for (auto& stream : m_streams)
-  {
-    if (&stream->m_adStream == adStream)
-    {
-      ISampleReader* streamReader = stream->GetReader();
-      if (!streamReader)
-        LOG::LogF(LOGWARNING, "Cannot get the stream sample reader");
-      else
-        streamReader->SetPTSOffset(stream->m_adStream.GetCurrentPTSOffset());
-
-      stream->segmentChanged = true;
-      break;
-    }
-  }
-}
-
-void Session::OnStreamChange(adaptive::AdaptiveStream* adStream)
-{
-  for (auto& stream : m_streams)
-  {
-    if (stream->enabled && &stream->m_adStream == adStream)
-    {
-      UpdateStream(*stream);
-      changed_ = true;
-    }
-  }
-}
-
-void Session::CheckFragmentDuration(STREAM& stream)
-{
-  uint64_t nextTs;
-  uint64_t nextDur;
-  ISampleReader* streamReader = stream.GetReader();
-  if (!streamReader)
-  {
-    LOG::LogF(LOGERROR, "Cannot get the stream sample reader");
-    return;
-  }
-
-  if (stream.segmentChanged && streamReader->GetNextFragmentInfo(nextTs, nextDur))
-  {
-    adaptiveTree_->SetFragmentDuration(
-        stream.m_adStream.getAdaptationSet(), stream.m_adStream.getRepresentation(),
-        stream.m_adStream.getSegmentPos(), nextTs, static_cast<uint32_t>(nextDur),
-        streamReader->GetTimeScale());
-  }
-  stream.segmentChanged = false;
-}
-
-const AP4_UI08* Session::GetDefaultKeyId(const uint16_t index) const
-{
-  static const AP4_UI08 default_key[16] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
-  if (adaptiveTree_->current_period_->psshSets_[index].defaultKID_.size() == 16)
-    return reinterpret_cast<const AP4_UI08*>(
-        adaptiveTree_->current_period_->psshSets_[index].defaultKID_.data());
-  return default_key;
-}
-
-Adaptive_CencSingleSampleDecrypter* Session::GetSingleSampleDecrypter(std::string sessionId)
-{
-  for (std::vector<CDMSESSION>::iterator b(cdm_sessions_.begin() + 1), e(cdm_sessions_.end());
-       b != e; ++b)
-    if (b->cdm_session_str_ && sessionId == b->cdm_session_str_)
-      return b->single_sample_decryptor_;
-  return nullptr;
-}
-
-uint32_t Session::GetIncludedStreamMask() const
-{
-  const INPUTSTREAM_TYPE adp2ips[] = {
-      INPUTSTREAM_TYPE_NONE, INPUTSTREAM_TYPE_VIDEO, INPUTSTREAM_TYPE_AUDIO,
-      INPUTSTREAM_TYPE_SUBTITLE};
-  uint32_t res(0);
-  for (unsigned int i(0); i < 4; ++i)
-    if (adaptiveTree_->current_period_->included_types_ & (1U << i))
-      res |= (1U << adp2ips[i]);
-  return res;
-}
-
-STREAM_CRYPTO_KEY_SYSTEM Session::GetCryptoKeySystem() const
-{
-  if (m_kodiProps.m_licenseType == "com.widevine.alpha")
-    return STREAM_CRYPTO_KEY_SYSTEM_WIDEVINE;
-#if STREAMCRYPTO_VERSION_LEVEL >= 1
-  else if (m_kodiProps.m_licenseType == "com.huawei.wiseplay")
-    return STREAM_CRYPTO_KEY_SYSTEM_WISEPLAY;
+#ifdef CreateDirectory
```

Comment:
this is unused

Reply:
For some reason this was needed back when adding using dllhelper, it wouldn't compile on some platforms. Seems to not be an issue now :)